tests/admin/views/test_rules.py

```diff
@@ -2238,6 +2238,146 @@ def testAddScheduledChangeMultipleConditions(self):
+            "version": "3.3",
+            "backgroundRate": 100,
+            "mapping": "c",
+            "update_type": "minor",
+            "data_version": 1,
+            "change_type": "insert",
+            "when": 1234567,
+            "base_alias": "TestDuplicateAlias1",
+        }
+        ret1 = self._post("/scheduled_changes/rules", data=data1)
+        self.assertEqual(ret1.status_code, 400, ret1.get_data())
+        load = ret1.get_json()
+        self.assertEqual(load["detail"], "Rule with alias exists")
+
+    @mock.patch("time.time", mock.MagicMock(return_value=300))
+    def testAddScheduledChangesRuleWithDuplicateAliasWithChangeTypeUpdate(self):
+        ret = self._post(
+            "/rules",
+            data=dict(backgroundRate=31, mapping="c", priority=33, product="Firefox", update_type="minor", channel="nightly", alias="TestDuplicateAlias2"),
+        )
+        self.assertEqual(ret.status_code, 200, "Status Code: %d, Data: %s" % (ret.status_code, ret.get_data()))
+
+        data1 = {
+            "rule_id": 10,
+            "telemetry_product": None,
+            "telemetry_channel": None,
+            "telemetry_uptake": None,
+            "priority": 80,
+            "buildTarget": "d",
+            "version": "3.3",
+            "backgroundRate": 100,
+            "mapping": "c",
+            "update_type": "minor",
+            "data_version": 1,
+            "change_type": "update",
+            "when": 1234567,
+            "base_alias": "TestDuplicateAlias2",
+        }
+        ret1 = self._post("/scheduled_changes/rules", data=data1)
+        self.assertEqual(ret1.status_code, 400, ret1.get_data())
+        load = ret1.get_json()
+        self.assertEqual(load["detail"], "Rule with alias exists")
+
+    @mock.patch("time.time", mock.MagicMock(return_value=300))
+    def testAddScheduledChangeWithAliasAlreadyPresentWithChangeTypeInsert(self):
+        data = {
+            "rule_id": 5,
+            "telemetry_product": None,
+            "telemetry_channel": None,
+            "telemetry_uptake": None,
+            "priority": 80,
+            "buildTarget": "d",
+            "version": "3.3",
+            "backgroundRate": 100,
+            "mapping": "c",
+            "update_type": "minor",
+            "data_version": 1,
+            "change_type": "insert",
+            "when": 1234567,
+            "base_alias": "TestDuplicateAlias3",
+            "complete": False,
+        }
+        ret = self._post("/scheduled_changes/rules", data=data)
+        self.assertEqual(ret.status_code, 200, ret.get_data())
+
+        data1 = {
+            "when": 2000000,
+            "data_version": 1,
+            "rule_id": 1,
+            "priority": 100,
+            "version": "3.5",
+            "buildTarget": "d",
+            "backgroundRate": 100,
+            "mapping": "c",
+            "update_type": "minor",
+            "sc_data_version": 1,
+            "base_alias": "TestDuplicateAlias3",
```

Comment:
Also try simulating a request like this on https://localhost:9000/rules/364 by adding an alias and clicking the Save button - take a dlook at what the the Request JSON looks like in the Developer Tools. If it doesn't use `base` neither should you in these tests.

Reply:
I have updated all the files to use `alias` instead of `base_alias`.

Reply:
You should be fully authorized in your local development environment...

In any case, I can assure you that the admin API doesn't use `base` in these requests.